typespec-tests/src/main/java/com/type/property/additionalproperties/models/ExtendsModelArrayAdditionalProperties.java

```diff
@@ -90,9 +91,14 @@ public static ExtendsModelArrayAdditionalProperties fromJson(JsonReader jsonRead
                 }
 
                 additionalProperties.put(fieldName, null);
+                if (reader.currentToken() == JsonToken.START_ARRAY) {
+                    additionalProperties.replace(fieldName, new ArrayList<>());
+                    while (reader.nextToken() != JsonToken.END_ARRAY) {
+                        additionalProperties.get(fieldName).add(ModelForRecord.fromJson(reader));
+                    }
+                }
```

Comment:
Would it work if the code here be

`additionalProperties.put(fieldName, reader.readArray(reader1 -> ModelForRecord.fromJson(reader1)));`

Reply:
It can work, now fixed in the new version.